packages/webapp/src/components/Animals/AddAnimalsFormCard/AddAnimalsFormCard.tsx

```diff
@@ -0,0 +1,144 @@
+/*
+ *  Copyright 2024 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+
+import { Controller, useFormContext } from 'react-hook-form';
+
+import NumberInput from '../../Form/NumberInput';
+import Checkbox from '../../Form/Checkbox';
+import SexDetails from '../../Form/SexDetails';
+import styles from './styles.module.scss';
+import Input from '../../Form/Input';
+import Card from '../../CardV2';
+import { Main } from '../../Typography';
+import TextButton from '../../Form/Button/TextButton';
+import { ReactComponent as XIcon } from '../../../assets/images/x-icon.svg';
+import { type Details as SexDetailsType } from '../../Form/SexDetails/SexDetailsPopover';
+import {
+  AnimalBreedSelect,
+  AnimalTypeSelect,
+  type AnimalBreedSelectProps,
+  type AnimalTypeSelectProps,
+  type Option,
+} from './AnimalSelect';
+import { useTranslation } from 'react-i18next';
+
+// Should be moved up to parent component that calls useForm
+type FormFields = {
+  type?: Option;
+  breed?: Option;
+  sexDetails: SexDetailsType;
+  count?: number;
+  createIndividualProfiles?: boolean;
+  group?: string;
+  batch?: string;
+};
+
+type AddAnimalsFormCardProps = AnimalTypeSelectProps &
+  AnimalBreedSelectProps & {
+    sexDetailsOptions: SexDetailsType;
+    onIndividualProfilesCheck?: (isChecked: boolean) => void;
+    onRemoveButtonClick?: (e: React.MouseEvent<HTMLButtonElement>) => void;
+    showRemoveButton?: boolean;
+    isActive?: boolean;
+  };
+
+export default function AddAnimalsFormCard({
+  typeOptions,
+  breedOptions,
+  sexDetailsOptions,
+  onTypeChange,
+  onIndividualProfilesCheck,
+  showRemoveButton,
+  onRemoveButtonClick,
+  isActive,
```

Comment:
Do you have an idea on how to decide if this card is active or not?

Reply:
I guess when any field inside the form is focused, a state variable `isActive` can be set to true

Reply:
Using a listener?? I'm wondering if we can use `:focus-within`...

Reply:
:focus-within might be good as well, or even better as it's just css